src/java/org/apache/cassandra/schema/MigrationCoordinator.java

```diff
@@ -467,16 +476,29 @@ void announce(UUID schemaVersion)
         SchemaDiagnostics.versionAnnounced(Schema.instance);
     }
 
-    private Future<Void> submitToMigrationIfNotShutdown(Runnable task)
+    private Future<?> submitToMigrationIfNotShutdown(Runnable task)
     {
-        if (executor.isShutdown() || executor.isTerminated())
+        boolean skipped = false;
+        try
         {
-            logger.info("Skipped scheduled pulling schema from other nodes: the MIGRATION executor service has been shutdown.");
+            if (executor.isShutdown() || executor.isTerminated())
+            {
+                skipped = true;
+                return ImmediateFuture.success(null);
+            }
+            return executor.submit(task);
+        }
+        catch (RejectedExecutionException ex)
+        {
+            skipped = true;
             return ImmediateFuture.success(null);
```

Comment:
shouldn't this return an failed future rather than success future?

Reply:
4.0 returned `null`.  The issue with this is that `executor.submit(task)` returns `Void`, which can only ever be `null`; so the caller has 0 way to know that the future was rejected; in 4.0 we could null check the future reference

Reply:
removing my comment due to the fact this looks to be the desired behavior of this function, so you are not changing anything, you are just being consistent with existing logic...